connection.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+
+function fastifyMysql (fastify, options, next) {
+  const usePromise = options.promise
+  delete options.promise
+
+  if (usePromise) {
+    _wrapCreateConnection(fastify, options, next)
+  } else {
+    const name = options.name
+    delete options.name
+
+    const mysql = require('mysql2')
+    const connection = mysql.createConnection(options.connectionString || options)
+    const db = {
+      connection: connection,
+      query: usePromise ? connection.query : connection.query.bind(connection),
+
+      // synchronous functions
+      format: mysql.format || connection.format,
+      escape: mysql.escape || connection.escape,
+      escapeId: mysql.escapeId || connection.escapeId
+    }
+    if (name) {
+      if (!fastify.mysql) {
+        fastify.decorate('mysql', {})
+      }
+      if (fastify.mysql[name]) {
+        next(new Error('fastify.mysql.' + name + 'has already registered'))
+      }
+      fastify.mysql[name] = db
+    } else {
+      if (fastify.mysql) {
+        next(new Error('fastify-mysql has already registered'))
+      } else {
+        fastify.mysql = db
+      }
+    }
```

Comment:
this block is identical in the two branches, this code should be shared (maybe inside index).